### PR TITLE
Create fetch_amazon_product_data.py

### PR DESCRIPTION
**Pull request by SparshRastogi**
This file provides a function which will take  
a product name as input from the user,and fetch the necessary   information about that kind of products from Amazon like the product  title,link to that product,price of the product,the ratings of  the product and the discount available on the product   in the form of a csv file,this will help the users by improving searchability   and navigability and find the right product easily and in a short period of time,  it will also be beneficial for performing better analysis on products

### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: web_programming/fetch_amazon_product_data.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+
+import itertools as it
+
+import pandas as pd
+import requests
+from bs4 import BeautifulSoup as bs
+
+
+def get_product_info(
```
[the remark this answers is not in the document]
Comment (SparshRastogi) [reply web_programming/fetch_amazon_product_data.py:18]: Pandas dataframe

File: web_programming/fetch_amazon_product_data.py [hunk 2]
```diff
@@ -15,17 +15,21 @@
...
 
 
-def get_product_info(product="laptop") -> None:
-    # function that will take the product as input and return the product details as output
-    # in the form of a csv file,if no input is given,it will fetch the details of laptop by default
+def get_product_info(product:str = "laptop") -> None:
+    # function that will take the product as input and return the 
+    #product details as output
+    # in the form of a csv file,if no input is given,it 
+    #will fetch the details of laptop by default
```
Comment (cclauss) [inline web_programming/fetch_amazon_product_data.py]:
```suggestion
    """
    Take the product as input and return the product details as output in the form
    of a csv file.  If no input is given, it will fetch the details of `laptop` by default.
```

**Comment by cclauss** [comment]
Here are the errors under the `Details` link next to the ❌ below:
```
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

web_programming/fetch_amazon_product_data.py:15:2: N813 camelcase 'BeautifulSoup' imported as lowercase 'bs'
web_programming/fetch_amazon_product_data.py:28:89: E501 line too long (129 > 88 characters)
web_programming/fetch_amazon_product_data.py:45:12: B007 Loop control variable 'j' not used within the loop body. If this is intended, start the name with an underscore.

mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

web_programming/fetch_amazon_product_data.py:82: error: Incompatible types in assignment (expression has type "str", variable has type "float")
Found 1 error in 1 file (checked 1092 source files)
```

**Comment by cclauss** [comment]
Please return data from your function instead of putting it into a csv file.  You can put that data into a csv somewhere else in this script.  The reason for doing this is explained in CONTRIBUTING.md.  The caller might want to do something else with the data (tweet it, or compare it or chart it over time or use it with IfThisThenThat) and your algorithm should not assume what that is.  We want to see that it can be stored in CSV but that action should be outside your algorithmic function.

File: web_programming/fetch_amazon_product_data.py [hunk 3]
```diff
@@ -42,7 +41,7 @@ def get_product_info(product: str = "laptop") -> None:
             "Discount",
         ]
     )  # initializing a pandas dataframe to store the requisite information
-    for i, j in it.zip_longest(
+    for i,j in it.zip_longest(
```
Comment (cclauss) [inline web_programming/fetch_amazon_product_data.py]: [psf/black](https://github.com/psf) is going to undo this change in `pre-commit`

Comment (SparshRastogi) [reply web_programming/fetch_amazon_product_data.py]: So,isn't there any way to use the two simultaneously? Or should I remove the second variable?

**Review by cclauss — APPROVED** [review]
I made changes based on the review comments.

**Comment by SparshRastogi** [comment]
@cclauss thank you so much for the edits to the file,actually I have just recently started contributing to open source,so I didn't had much idea about that but after seeing your edits I got to learn a lot and it will be helpful for me while  contributing in future
